codegen/types.go

```diff
@@ -108,6 +108,48 @@ var (
 	}
 )
 
+// GoTypeDef returns the Go code that defines a Go type which matches the data structure
+// definition (the part that comes after `type foo`).
+func GoTypeDef(dt design.DataType) string {
+	switch actual := dt.(type) {
+	case design.Primitive:
+		return GoTypeName(actual)
+	case *design.Array:
+		d := GoTypeDef(actual.ElemType.Type)
+		if design.IsObject(actual.ElemType.Type) {
+			d = "*" + d
+		}
+		return "[]" + d
+	case *design.Map:
+		keyDef := GoTypeDef(actual.KeyType.Type)
+		if design.IsObject(actual.KeyType.Type) {
+			keyDef = "*" + keyDef
+		}
+		elemDef := GoTypeDef(actual.ElemType.Type)
+		if design.IsObject(actual.ElemType.Type) {
+			elemDef = "*" + elemDef
+		}
+		return fmt.Sprintf("map[%s]%s", keyDef, elemDef)
+	case design.Object:
+		return goTypeDefObject(actual)
+	case *design.UserTypeExpr:
```

Comment:
Please use `design.UserType` and not `design.UserTypeExpr` so that it also works for media types. Would be good to have tests for this function that tests all the different types (primitive, array, map, object, user and media types).